plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt

```diff
@@ -335,45 +337,89 @@ class SsoAccessTokenProvider(
-                AwsTelemetry.refreshCredentials(
-                    project = null,
-                    result = Result.Failed,
-                    sessionDuration = sessionDuration.toHours().toInt(),
-                    credentialSourceId = credentialSourceId,
-                    reason = "Null refresh token"
-                )
-            }
-
-            throw InvalidRequestException.builder().message("Requested token refresh, but refresh token was null").build()
+            val message = "Requested token refresh, but refresh token was null"
+            sendFailedRefreshCredentialsMetricIfNeeded(
+                currentToken,
+                reason = "Null refresh token",
+                reasonDesc = message
+            )
+            throw InvalidRequestException.builder().message(message).build()
         }
 
         val registration = when (currentToken) {
             is DeviceAuthorizationGrantToken -> loadDagClientRegistration()
             is PKCEAuthorizationGrantToken -> loadPkceClientRegistration()
-        } ?: throw InvalidClientException.builder().message("Unable to load client registration").build()
-
-        val newToken = client.createToken {
-            it.clientId(registration.clientId)
-            it.clientSecret(registration.clientSecret)
-            it.grantType(REFRESH_GRANT_TYPE)
-            it.refreshToken(currentToken.refreshToken)
         }
-
-        val token = when (currentToken) {
-            is DeviceAuthorizationGrantToken -> newToken.toDAGAccessToken(currentToken.createdAt)
-            is PKCEAuthorizationGrantToken -> newToken.toPKCEAccessToken(currentToken.createdAt)
+        if (registration == null) {
+            val message = "Unable to load client registration"
+            sendFailedRefreshCredentialsMetricIfNeeded(
+                currentToken,
+                reason = "Null client registration",
+                reasonDesc = message
+            )
+            throw InvalidClientException.builder().message(message).build()
         }
 
-        saveAccessToken(token)
+        try {
+            val newToken = client.createToken {
+                it.clientId(registration.clientId)
+                it.clientSecret(registration.clientSecret)
+                it.grantType(REFRESH_GRANT_TYPE)
+                it.refreshToken(currentToken.refreshToken)
+            }
 
-        return token
+            val token = when (currentToken) {
+                is DeviceAuthorizationGrantToken -> newToken.toDAGAccessToken(currentToken.createdAt)
+                is PKCEAuthorizationGrantToken -> newToken.toPKCEAccessToken(currentToken.createdAt)
+            }
+
+            saveAccessToken(token)
+
+            return token
+        } catch (e: Exception) {
+            val requestId = when (e) {
+                is SsoOidcException -> e.requestId()
```

Comment:
doesnt `SsoOidcException` extend `AwsServiceException`?

Reply:
Yes it does. Fixing